### PR TITLE
Add Device Type Attribute

### PR DESCRIPTION
**Pull request by nishadg246**
Add tag to each UrlDevice defining the tranport it was discovered with.

### CODE REVIEW

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java [hunk 1]
```diff
@@ -173,6 +174,7 @@ private void restoreCache() {
 
     // Don't load the cache if it's stale
     mScanStartTime = prefs.getLong(SCAN_START_TIME_KEY, 0);
+    prevScanStartTime = mScanStartTime;
     if (now - mScanStartTime >= SCAN_STALE_TIME_MILLIS) {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java]: just compute `now - mScanStartTime` once.

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java [hunk 2]
```diff
@@ -189,6 +191,13 @@ private void restoreCache() {
           Utils.updateRegion(urlDevice);
         }
       }
+      if (now - prevScanStartTime >= LOCAL_SCAN_STALE_TIME_MILLIS) {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java]: why does this need to go in the try block?

Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java]: Comment this section

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java [hunk 3]
```diff
@@ -183,17 +185,26 @@ private void restoreCache() {
...
     } catch (PhysicalWebCollectionException e) {
       Log.e(TAG, "Could not restore Physical Web collection cache", e);
     }
+    // replace TxPower and RSSI data after restoring cache
+    for (UrlDevice urlDevice : mPwCollection.getUrlDevices()) {
+      if (Utils.isBleUrlDevice(urlDevice)) {
+        Utils.updateRegion(urlDevice);
+      }
+    }
+    // if the unresolvable devices cache is
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java]: Ideally, use complete sentences.  Say something like "Unresolvable devices are typically not relevant outside of scan range.  Hence, we specially clean them from the cache."

**Comment by cco3** [comment]
LGTM

**Comment by nondebug** [comment]
lgtm

**Pull request merged**